text/0178-mapping-to-otlp-anyvalue.md

```diff
@@ -0,0 +1,227 @@
+field using decimal representation.
+
+#### Enumerations
+
+Values, which belong to a limited enumerated set (e.g. a Java
+[enum](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)) SHOULD be
+converted to AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
+field with the value of the string set to the symbolic name of the enumeration.
+
+If the symbolic name of the enumeration is not possible to obtain the
+implementation SHOULD map enumeration's value to AnyValue's
+[int_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L33)
+field set equal to enum's ordinal number when such ordinal number is naturally
+obtainable.
+
+If the ordinal value is also not possible to obtain the enumeration SHOULD be
+converted to AnyValue's
+[bytes_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L37)
+field in any manner that the implementation deems reasonable.
+
+#### Floating Point Values
+
+Floating point values which are within the range and precision of IEEE 754
+64-bit floating point numbers (including IEEE 32-bit floating point values)
+SHOULD be converted to AnyValue's
+[double_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L34)
+field.
+
+Floating point values which are outside the range or precision of IEEE 754
+64-bit floating point numbers (e.g. IEEE 128-bit floating bit values) SHOULD be
+converted to AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
+field using decimal floating point representation.
+
+#### String Values
+
+String values which are valid Unicode sequences SHOULD be converted to
+AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
+field.
+
+String values which are not valid Unicode sequences SHOULD be converted to
+AnyValue's
+[bytes_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L37)
+with the bytes representing the string in the original order and format of the
+source string.
+
+#### Bytes Sequences
+
+Byte sequences (e.g. Go's `[]byte` slice or raw byte content of a file) SHOULD
+be converted to AnyValue's
+[bytes_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L37)
+field.
+
+### Composite Values
```

Comment:
It might be prudent to explicitly call out at this level of detail that composite values can contain values of differing types, though it also does seem clear from the type definitions.